guides/session.rst

```diff
@@ -1,76 +1,116 @@
-browser emulator with it:
+The first thing to do when using your session is to open a page with it.
+Just after starting, the session is not on any page (in a real browser, you
+would on the ``about:blank`` page), and calling any other action is likely
+to fail.
 
 .. code-block:: php
 
-    // open some page in browser:
     $session->visit('http://my_project.dev/some_page.php');
 
-    // get the current page URL:
-    echo $session->getCurrentUrl();
+.. note::
 
-    // get the response status code:
-    echo $session->getStatusCode();
+    Mink is primarily designed to be used for testing websites. To allow
+    covering error pages as well, the ``Session::visit`` method does not
+    consider error status codes as invalid. It will not throw an exception
+    in this case. You will need to check whether the response was a success
+    or an error. It will only throw an exception when Mink cannot load the
+    page (network error, ...). For cases, when driver does not support status
+    code retrieval, it might be possible to assert text on the page itself
+    to determine whether it is an error page or no.
 
-    // get page content:
-    echo $session->getPage()->getContent();
+Interacting with the Page
+-------------------------
 
-    // open another page:
-    $session->visit('http://my_project.dev/second_page.php');
+The session gives you access to the page through the ``Session::getPage``
+method. This allows you to :doc:`traverse </guides/traversing-pages>` and
+:doc:`manipulate it </guides/manipulating-pages>`. The next chapters are
+covering the page API in depth.
+
+Using the Browser History
+-------------------------
+
+The session gives you access to the browser history:
+
+.. code-block:: php
+
+    // get the current page URL:
+    echo $session->getCurrentUrl();
 
     // use history controls:
     $session->reload();
     $session->back();
     $session->forward();
 
-    // evaluate JS expression:
-    echo $session->evaluateScript(
-        "return 'something from browser';"
-    );
+Cookies Management
+------------------
 
-    // wait for n milliseconds or
-    // till JS expression becomes true:
-    $session->wait(
-        5000,
-        "$('.suggestions-results').children().length > 0"
-    );
+The session can manipulate cookies available in the browser.
+
+.. code-block:: php
+
+    // set cookie:
+    $session->setCookie('cookie name', 'value');
+
+    // get cookie:
+    echo $session->getCookie('cookie name');
+
+    // delete cookie:
+    $session->setCookie('cookie name', null);
 
 .. note::
 
-    Although Mink does its best on removing browser differences between different
-    browser emulators, it can't do much in some cases. See the :ref:`driver-feature-support`
-    to see which features are supported by each driver.
+    In browser controllers, you may be be restricted to controlling only
+    http-only cookies because others cannot be accessed through JavaScript.
```

Comment:
I wasn't clear on "browser controllers" - can we say "With drivers that use a browser - like Selenium2 - you may be restricted to accessing/setting `HttpOnly` cookies...

With a link on HttpOnly maybe to http://en.wikipedia.org/wiki/HTTP_cookie#HttpOnly_cookie - I'll admit that it's not a property on a Cookie I think about much.